### DISCUSSION TITLE
[Feature Request] Add a callback like formatMessageCB for every step

### DISCUSSION THREAD

**Discussion by SuiKaSan** [Zhengqbbb/cz-git#209] [Announcements]
### 💭 Describe the feature

Add a callback like formatMessageCB for every step.
Like if I selected a **scope**, then triggers a callback to customize the type or the following subject.

In my case, I need to write a commit meesage for every version release with a scope called version.
So I have to type "release version xxx" every time I wirte a version related commit.
It's annoying but I can't just left it empty.

**defaultSubject** will fix the problem but everyone write a commit would see the default subject, that's not good.

### 💡 Proposed Solution

A callback to every step would be nice, so the subject can be altered.
Like if I know the coming scope is **version**, then change the **defaultSubject** to "release version ", if not, then leave it empty.

It's just a feature nice to have.
And thank you very much for developing this library, it's crazy good!

**Comment by Zhengqbbb** [comment]
e.g ?
```txt
build(version): release xxx version
```

Each step provides a callback Fn, Emmm I will not add similar configurations for special requirements to avoid increasing complexity. 
You can provide some examples and cases for us to analyze and see if we can handle them well with the existing configurations.

---

> release version xxx

1. Replace scope based on special type

https://cz-git.qbb.sh/config/engineer#scopeoverrides
👇
type select `build` => only scope `version` => type `xxx` => using callback Fn gen template message `build(version): release xxx version`

2. I wonder if this kind of commit is only needed when there is a release version, so this scenario should be other CLI like `bumpp`, rather than using CLI `cz` `czg`.

https://github.com/Zhengqbbb/cz-git/blob/328fb2ef94b71ab3daf47db4bdda17114ea5dab6/package.json#L44-L45

![CleanShot-2024-12-26_17_55_51@2x](https://github.com/user-attachments/assets/4cae977e-7500-4d7e-87ba-5bc029c3cb1c)
